target/mips/helper.c

```diff
@@ -1762,6 +1762,19 @@ static inline ram_addr_t v2r_addr(CPUMIPSState *env, target_ulong vaddr, MMUAcce
     return ram_addr;
 }
 
+static inline void cheri_linkedflag_invalidate(hwaddr paddr, int32_t size)
+{
+    CPUState *other_cs = first_cpu;
+
+    CPU_FOREACH(other_cs) {
+        MIPSCPU *other_cpu = MIPS_CPU(other_cs);
+        CPUMIPSState *other_env = &other_cpu->env;
+        /* Check RAM address to see if the linkedflag needs to be reset. */
+        if (paddr <= other_env->lladdr && other_env->lladdr < paddr + size)
+            other_env->linkedflag = 0;
+    }
```

Comment:
We should really move away from using the linkedflag hack. But that can be done some other time.
The next time we merge from upstream, the ll/sc handling in the MIPS target will have changed so we should probably use the same approach for CHERI atomics.

Reply:
Have a convenient pointer for what upstream does now?

Reply:
Hmm we might actually have merged this already. 
Anyway, linkedflag was added only for CHERI and means that CHERI cll* and MIPS ll* use separate linked statuses so I believe we can have both ll/sc and cllc/cscc succeed even if they touch the same address...